src/main/java/software/amazon/awssdk/s3accessgrants/plugin/S3AccessGrantsAuthSchemeProvider.java

```diff
@@ -93,10 +94,20 @@ public List<AuthSchemeOption> resolveAuthScheme(@NotNull S3AuthSchemeParams auth
             logger.debug(() -> "Access Grants requests will be sent to the region "+destinationRegion);
 
             String S3Prefix;
-            if (operation.equalsIgnoreCase("DELETEOBJECTS")){
+            if (operation.equalsIgnoreCase("DELETEOBJECTS")) {
                 S3Prefix = "s3://" + authSchemeParams.bucket() + getCommonPrefixFromMultiplePrefixes(authSchemeParams.deleteObjectKeys());
             }
-            else{
+            else if (operation.equalsIgnoreCase("COPYOBJECT")) {
+                String[] copySourceArray = authSchemeParams.copySource().split("/", 2);
+                if (!copySourceArray[0].equals(authSchemeParams.bucket())) {
+                    throw SdkServiceException.builder().message("The requested operation cannot be completed!").statusCode(404).cause(new RuntimeException("Source bucket and destination bucket must be same")).build();
+                }
+                List<String> copyObjectKeys = new ArrayList<>();
+                copyObjectKeys.add(authSchemeParams.key());
```

Comment:
I am wondering what happens if key is returned as a NULL object or what happens if both the keys you are inserting in copyObjectKeys are null? would getCommonPrefixFromMultiplePrefixes return '/' as the result? I think it would be a good idea to add some unit tests around this. Ignore if you already have unit tests covering these cases.

Reply:
yes getCommonPrefixFromMultiplePrefixes return / in that case. We already have a unit test for this.